compiler/main/checks.cpp

```diff
@@ -700,6 +703,86 @@ checkAutoCopyMap()
   }
 }
 
+static FnSymbol* findUserInitEq(AggregateType* at) {
+  for_alive_in_Vec(FnSymbol, fn, gFnSymbols) {
+    if (fn->name == astrInitEquals &&
+        !fn->hasFlag(FLAG_COMPILER_GENERATED) &&
+        fn->numFormals() >= 2) {
+      ArgSymbol* lhs = fn->getFormal(1);
+      Type* t = lhs->getValType();
+      if (t == at)
+        return fn;
+    }
+  }
+  return NULL;
+}
+
+static FnSymbol* findUserAssign(AggregateType* at) {
+
+  for_alive_in_Vec(FnSymbol, fn, gFnSymbols) {
+    if (fn->name == astrSassign &&
+        !fn->hasFlag(FLAG_COMPILER_GENERATED) &&
+        fn->numFormals() >= 1) {
+      ArgSymbol* lhs = fn->getFormal(1);
+      Type* t = lhs->getValType();
+      if (t == at)
+        return fn;
+    }
+  }
+  return NULL;
+}
+
+
+static void
+checkDefaultInitEqAndAssign()
+{
+  for_alive_in_Vec(TypeSymbol, ts, gTypeSymbols) {
+    if (ts->hasFlag(FLAG_EXTERN))
+      continue; // can't make init= for extern types today anyway
+    if (!isRecord(ts->type) && !isUnion(ts->type))
+      continue; // can't make init= for non-records today anyway
+
+    AggregateType* at = toAggregateType(ts->type);
+    bool defaultInitEq = ts->hasFlag(FLAG_TYPE_DEFAULT_INIT_EQUAL);
+    bool customInitEq = ts->hasFlag(FLAG_TYPE_CUSTOM_INIT_EQUAL);
+    bool defaultAssign = ts->hasFlag(FLAG_TYPE_DEFAULT_ASSIGN);
+    bool customAssign = ts->hasFlag(FLAG_TYPE_CUSTOM_ASSIGN);
+    if (defaultInitEq && customAssign) {
+      USR_FATAL_CONT(ts, "Type '%s' uses compiler-generated default init= "
+                         "but has a custom = function. "
+                         "Please add an init= method",
+                         toString(ts->type));
```

Comment:
Should these error messages have a ` around the init= and = portions of it?

Reply:
I put single quotes like other error messages. It's not obvious to me that's better but I interpreted your comment as a suggestion.